server/src/storage/types.rs

```diff
@@ -150,7 +132,279 @@ impl SqlType {
+                    },
+                    CompType::SEThan => {
+                        self.compare_byte_greater_than(val, val2).map(|x| !x)
+                    },
+                }
+            },
+        }
+    }
+    /// fn compares slices of u8 byte for byte and returns if both values are equal
+    /// returns boolean on success and Error when given values do not have the same size
+    fn compare_byte_for_equal(&self, val: &[u8], val2: &[u8])
+    -> Result<bool, Error>
+    {
+        if val.len() != val2.len() {
+            return Err(Error::WrongLength)
+        }
+        info!("start comparing each byte");
+        for i in 0 .. val.len() {
+            if val[i] != val2[i] {
+                return Ok(false)
+            }
+        }
+        Ok(true)
+    }
+    /// fn compares slices of u8 byte for byte and returns
+    /// if first given value is greater than the second one
+    /// returns boolean on success and Error when given values do not have the same size
+    fn compare_byte_greater_than(&self, val: &[u8], val2: &[u8])
+    -> Result<bool, Error>
+    {
+        info!("start comparing each byte");
+        if val.len() != val2.len() {
+            return Err(Error::WrongLength)
+        }
+        for i in 0 .. val.len() {
+            if val[i] > val2[i] {
+                return Ok(true)
+            }
+        }
+        Ok(false)
+    }
+
+    /// fn compares slices of u8 byte for byte and returns
+    /// if first given value is lesser than the second one
+    /// returns boolean on success and Error when given values do not have the same size
+    fn compare_byte_lesser_than(&self, val: &[u8], val2: &[u8])
+    -> Result<bool, Error>
+    {
+        info!("start comparing each byte");
+        if val.len() != val2.len() {
+            return Err(Error::WrongLength)
+        }
+        for i in 0 .. val.len() {
+            if val[i] < val2[i] {
+                return Ok(true)
+            }
+        }
+        Ok(false)
+    }
+
+    /// fn compares slices of u8 as booleans and returns
+    /// if both both booleans are true
+    /// returns boolean on success and Error when given values do not have the same size
+    fn compare_as_bool(&self, val: &[u8], val2: &[u8])
+    -> Result<bool, Error>
+    {
+        info!("start converting to bool");
+        let mut sov: bool = false;
+        let mut sov2: bool = false;
+        for i in 0 .. val.len() {
+            if val[i] != 0 {
+                sov = true;
+                break;
+            }
+        }
+        info!("defined value 1 as bool: {:?}",sov);
+        for i in 0 .. val2.len() {
+            if val2[i] != 0 {
+                sov2 = true;
+                break;
+            }
+        }
+        info!("defined value 2 as bool: {:?}",sov2);
+        info!("start comparing bool");
+        if sov == sov2 {
+            return Ok(true)
+        }
+
+        Ok(false)
+    }
```

Comment:
Just return an error if one of those slices is longer than one byte. Booleans are always one byte wide, so check length and compare with `==`